contracts/UniswapV2Pair.sol

```diff
@@ -68,6 +69,13 @@ contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
         token0 = _token0;
         token1 = _token1;
     }
+    
+    // called by the factory to set the swapFee
+    function setSwapFee(uint8 _swapFee) external {
+        require(msg.sender == factory, 'UniswapV2: FORBIDDEN'); // sufficient check
+        require(_swapFee <= 30, 'UniswapV2: FORBIDDEN_FEE'); // fee percentage check
```

Comment:
Don't we want a minimum swap fee of 0.01%? i.e. `require(1 <= _swapFee <= 30)`?

Reply:
Good question, we talked about this with the rest of the team and we realized that there is no harm in allowing the swapFee to be 0% (at your own risk!).
We expect that there will be cases where the token pair owners (they can be project owners, traders, any individual with liquidity who is only interested in providing liquidity without revenue) would want to provide the liquidity itself and have a 0% fee. Crazy right? but we think that something like this can happen, even in the case of DXD, where there is already 100k pre-minted tokens, so the DXdao has plenty of liquidity to support a 0% swapFee token pair.
Also there are test cases for the 0% fee in the tests and it mints 0% of liquidity in mint/burn as expected.

Reply:
0 fees can be interesting (for example setting up prediction markets in order to gain information and not make profit).
Do we want a maximum fee? And specially such a low one? To continue on the prediction market example, most prediction markets a 0.3% fee seems way too low for liquidity providing to be profitable.
But I guess there is some advantage in guaranteeing that user never get charged crazy fees. However I'm not sure it's the role of the smart contract. The UI could do it.

Reply:
> But I guess there is some advantage in guaranteeing that user never get charged crazy fees. However I'm not sure it's the role of the smart contract. The UI could do it.

The problem I have with this is when interacting with contracts manually, or through a dex aggregator for example, we can't ensure user awareness of fees.

Reply:
you can ensure that there is going to be a maximum fee. And you can access the fee at anytime at the smart contract level too.